tests/test_autonomy/test_cli/test_fetch.py

```diff
@@ -86,18 +86,41 @@ def test_fetch_service_local(
 
         shutil.rmtree(service)
 
-    def test_fetch_service_ipfs(
+    def test_publish_and_fetch_service_ipfs(
         self,
     ) -> None:
         """Test fetch service."""
+        expected_hash = "bafybeidwo3rj7axx7l7p7vc52vm6hh2v7n6tqxly6om4gmhpnddbz7wmji"
 
-        service = self.t / "counter"
-        service_hash = self.get_service_hash()
+        service_dir = self.t / "dummy_service"
+        service_file = service_dir / "service.yaml"
+        service_dir.mkdir()
+        with open(service_file, "w+") as fp:
+            yaml.dump_all(get_dummy_service_config(), fp)
+
+        os.chdir(service_dir)
+
+        with mock.patch(
+            "autonomy.cli.publish.get_default_remote_registry", new=lambda: "ipfs"
+        ), mock.patch(
+            "autonomy.cli.publish.get_ipfs_node_multiaddr",
+            new=lambda: IPFS_REGISTRY,
+        ), mock.patch(
+            "click.echo"
+        ) as echo_mock:
+            result = self.cli_runner.invoke(cli, ["publish", "--remote"])
+            output = echo_mock.call_args[0][0]
+
+            assert result.exit_code == 0, output
+            assert expected_hash in output, output
```

Comment:
We publish a dummy service and fetch it later rather than fetching counter service. This we way test both publish and fetch commands without spamming the IPFS registry every time a small change occurs on the counter service.